### PR TITLE
#660: Feature request: configurable spaces around infix types

### PR DESCRIPTION
**Pull request by ysusuk**

### CODE REVIEW

File: core/src/main/scala/org/scalafmt/config/Spaces.scala [hunk 1]
```diff
@@ -6,5 +6,6 @@ import metaconfig.ConfigReader
 case class Spaces(
     beforeContextBoundColon: Boolean = false,
     afterTripleEquals: Boolean = false,
-    inImportCurlyBraces: Boolean = false
+    inImportCurlyBraces: Boolean = false,
+    neverAroundInfixTypes: Seq[String] = Seq("##")
```
Comment (olafurpg) [inline core/src/main/scala/org/scalafmt/config/Spaces.scala]: Could we keep the default at `Nil`? The `##` operator is only magical if you're using a specific compiler plugin.

**Comment by olafurpg** [comment]
Two more things I forgot

- Could you add a test with `##` appearing in Term.ApplyInfix position? Example `val x: F[X] ## B = f##b` should format as `val x: F[X]##B = f ## b`. The option should only affect infix **type** operators.
- Could you document the new option in the `Spaces` docstring?

File: core/src/main/scala/org/scalafmt/config/Spaces.scala (hunk 1 above)
Comment (ysusuk) [inline core/src/main/scala/org/scalafmt/config/Spaces.scala]: Agree! I was expecting this ^_^ but was not sure what is better as default value.

File: core/src/main/scala/org/scalafmt/internal/FormatOps.scala [hunk 2]
```diff
@@ -474,10 +474,14 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       token <- arg.tokens.lastOption
     } yield token).getOrElse(owner.tokens.last)
 
-    if (style.spaces.neverAroundInfixTypes.contains((op.value)))
-      Split(NoSplit, 0)
-    else
-      Split(modification, 0).withIndent(Num(indent), expire, ExpiresOn.Left)
+    owner.parent match {
```
Comment (ysusuk) [inline core/src/main/scala/org/scalafmt/internal/FormatOps.scala]: not sure that such an construct clearly defines an intention.

**Comment by ysusuk** [comment]
@olafurpg not sure what did you mean by `Spaces` docsring =(. I added param description in `ScalafmtConfig`

**Comment by olafurpg** [comment]
I meant to create a docstring in spaces, but this is fine too. Need to move the scalafmtconfig docstrings to the nested values some, never got around to it after introducing the hierarchy. 

LGTM 👍 thanks!

**Pull request merged**